admiral/pkg/controller/admiral/globaltraffic_test.go

```diff
@@ -25,3 +29,118 @@ func TestNewGlobalTrafficController(t *testing.T) {
+	}
+
+	gtpName := "gtp1"
+	gtp := model.GlobalTrafficPolicy{Selector: map[string]string{"identity": "payments", "env": "e2e"}, Policy:[]*model.TrafficPolicy{}}
+	gtpObj := makeK8sGtpObj(gtpName, "namespace1", gtp)
+	globalTrafficController.Added(gtpObj)
+
+	updatedGtp := model.GlobalTrafficPolicy{Selector: map[string]string{"identity": "payments", "env": "qa"}, Policy:[]*model.TrafficPolicy{}}
+	updatedGtpObj := makeK8sGtpObj(gtpName, "namespace1", updatedGtp)
+
+	globalTrafficController.Updated(updatedGtpObj, gtpObj)
+}
+
+func TestGlobalTrafficGetGtps(t *testing.T) {
+	config, err := clientcmd.BuildConfigFromFlags("", "../../test/resources/admins@fake-cluster.k8s.local")
+	if err != nil {
+		t.Errorf("%v", err)
+	}
+	stop := make(chan struct{})
+	handler := test.MockGlobalTrafficHandler{}
+
+	globalTrafficController, err := NewGlobalTrafficController(stop, &handler, config, time.Duration(1000))
+
+	globalTrafficController.CrdClient = fake.NewSimpleClientset()
+
+	if err != nil {
+		t.Errorf("Unexpected err %v", err)
+	}
+
+	if globalTrafficController == nil {
+		t.Errorf("GlobalTraffic controller should never be nil without an error thrown")
+	}
+
+	gtps, _  := globalTrafficController.GetGlobalTrafficPolicies()
+
+	if gtps != nil {
+		t.Errorf("gtps is not nil")
+	}
+
+	gtpName := "gtp1"
+	gtp := model.GlobalTrafficPolicy{Selector: map[string]string{"identity": "payments", "env": "e2e"}, Policy:[]*model.TrafficPolicy{}}
+	gtpObj := makeK8sGtpObj(gtpName, "namespace1", gtp)
+
+	globalTrafficController.CrdClient.AdmiralV1().GlobalTrafficPolicies("namespace1").Create(gtpObj);
+
+	gtps, err  = globalTrafficController.GetGlobalTrafficPolicies()
+
+	if err != nil {
+		t.Errorf("get gtps shouldn't error out, got: %v", err)
+	}
+
+	if gtps == nil || len(gtps) == 0 {
+		t.Errorf("gtps should be non empty")
+	}
+}
+
+func TestGlobalTrafficGetByLabel(t *testing.T) {
+	config, err := clientcmd.BuildConfigFromFlags("", "../../test/resources/admins@fake-cluster.k8s.local")
+	if err != nil {
+		t.Errorf("%v", err)
+	}
+	stop := make(chan struct{})
+	handler := test.MockGlobalTrafficHandler{}
+
+	globalTrafficController, err := NewGlobalTrafficController(stop, &handler, config, time.Duration(1000))
+
+	if err != nil {
+		t.Errorf("Unexpected err %v", err)
+	}
+
+	if globalTrafficController == nil {
+		t.Errorf("GlobalTraffic controller should never be nil without an error thrown")
+	}
+
+	gtps  := globalTrafficController.GetGTPByLabel("payments", "namespace1")
+
+	if gtps != nil || len(gtps) > 0 {
+		t.Errorf("gtps is not empty")
+	}
+
+	gtpName := "gtp1"
+	gtp := model.GlobalTrafficPolicy{Selector: map[string]string{"identity": "payments", "env": "e2e"}, Policy:[]*model.TrafficPolicy{}}
+	gtpObj := makeK8sGtpObj(gtpName, "namespace1", gtp)
+
+	globalTrafficController.CrdClient.AdmiralV1().GlobalTrafficPolicies("namespace1").Create(gtpObj);
+
+	gtps  = globalTrafficController.GetGTPByLabel("payments", "namespace1")
+
+	if gtps == nil || len(gtps) == 0 {
```

Comment:
Can we assert the objects match?

Reply:
The fake client doesn't seem to work, will fix it as part of the CRD group migration. Issue filed.